File: backend/app/modules/auth/controller.py
from flask import Blueprint, jsonify, request

from app.modules.auth.service import AuthService
from app.routes.auth import login_required

auth_bp = Blueprint("auth", __name__, url_prefix="/api/auth")
service = AuthService()


@auth_bp.post("/login")
def login():
    return jsonify(service.login(request.get_json(force=True))), 200


@auth_bp.get("/me")
@login_required()
def me(current_user):
    return jsonify({"user": current_user}), 200

File: backend/src/modules/auth/controller.ts
import type { Express } from "express";

import { asyncHandler } from "../../core/http";
import { loginRequired } from "../../routes/auth";
import { AuthService } from "./service";

const service = new AuthService();

export function registerAuthRoutes(app: Express) {
  app.post(
    "/api/auth/login",
    asyncHandler(async (req, res) => {
      res.status(200).json(await service.login(req.body ?? {}));
    }),
  );

  app.get(
    "/api/auth/me",
    loginRequired(),
    asyncHandler(async (_req, res) => {
      res.status(200).json({ user: res.locals.currentUser });
    }),
  );
}
